src/adapter/src/coord/sequencer/inner/explain_timestamp.rs

```diff
@@ -0,0 +1,335 @@
+    fn explain_timestamp_optimize(
+        &self,
+        ExplainTimestampOptimize {
+            validity,
+            plan,
+            cluster_id,
+        }: ExplainTimestampOptimize,
+    ) -> Result<StageResult<Box<ExplainTimestampStage>>, AdapterError> {
+        // Collect optimizer parameters.
+        let optimizer_config = optimize::OptimizerConfig::from(self.catalog().system_config());
+
+        // Build an optimizer for this VIEW.
+        let mut optimizer = optimize::view::Optimizer::new(optimizer_config, None);
+
+        let span = Span::current();
+        Ok(StageResult::Handle(mz_ore::task::spawn_blocking(
+            || "optimize explain timestamp",
+            move || {
+                span.in_scope(|| {
+                    let plan::ExplainTimestampPlan {
+                        format,
+                        raw_plan,
+                        when,
+                    } = plan;
+
+                    // HIR ⇒ MIR lowering and MIR ⇒ MIR optimization (local)
+                    let optimized_plan = optimizer.optimize(raw_plan)?;
+
+                    let stage =
+                        ExplainTimestampStage::RealTimeRecency(ExplainTimestampRealTimeRecency {
+                            validity,
+                            format,
+                            optimized_plan,
+                            cluster_id,
+                            when,
+                        });
+                    Ok(Box::new(stage))
+                })
+            },
+        )))
+    }
+
+    #[instrument]
+    async fn explain_timestamp_real_time_recency(
+        &mut self,
+        session: &Session,
+        ExplainTimestampRealTimeRecency {
+            validity,
+            format,
+            optimized_plan,
+            cluster_id,
+            when,
+        }: ExplainTimestampRealTimeRecency,
+    ) -> Result<StageResult<Box<ExplainTimestampStage>>, AdapterError> {
+        let source_ids = optimized_plan.depends_on();
+        let fut = self
+            .determine_real_time_recent_timestamp(session, source_ids.iter().cloned())
+            .await?;
+
+        match fut {
+            Some(fut) => {
+                let span = Span::current();
+                Ok(StageResult::Handle(mz_ore::task::spawn(
+                    || "explain timestamp real time recency",
+                    async move {
+                        let real_time_recency_ts = fut.await?;
+                        let stage = ExplainTimestampStage::Finish(ExplainTimestampFinish {
+                            validity,
+                            format,
+                            optimized_plan,
+                            cluster_id,
+                            source_ids,
+                            when,
+                            real_time_recency_ts: Some(real_time_recency_ts),
+                        });
+                        Ok(Box::new(stage))
+                    }
+                    .instrument(span),
```

Comment:
Uncovered according to the [coverage build,](https://buildkite.com/materialize/coverage/builds/427) please consider adding a test.

Reply:
I verified with a println that this is already tested in testdrive at https://github.com/MaterializeInc/materialize/blob/be6c424a7280eabf8c99dbc833808ac24e5b3791/test/testdrive/explain-timestamps.td#L27. Does the coverage build use that? Am I missing something else?